examples/cpp_for_opencl/default.cl

```diff
@@ -0,0 +1,8 @@
+template<typename T>
```

Comment:
Should we rename this to `default.clcpp`, add `.clcpp` to the file extensions list, and compile with `-x cl` on versions that don't recognise the `.clcpp` extension?

Reply:
I'm in no position to answer this; happy with whatever you all think makes the most sense :)

Reply:
Yes this makes sense. Also if you pass `-cl-std=clc++` it will activate C++ for OpenCL.

Reply:
Passing `-cl-std=clc++` is presumably necessary anyway for older Clang versions that lack support for the `.clcpp` file extension?  Is `-x cl` still needed when using the `.clcpp` file extension together with `-cl-std=clc++` for those older Clang versions?

I'm also happy with whatever makes the most sense, but I think it might make sense to use the `.clcpp` file extension.

Reply:
Yes, it is needed - I did a quick test and if the file extension is `.clcpp`, both `-cl-std=clc++` and `-x cl` are needed for older Clang versions.